cli/tests/lang-node.bats

```diff
@@ -91,23 +91,37 @@ teardown() {
     *-linux)
       # Ensure we're getting krb5 from the flox package by first checking
       # installation fails
-      run ! "$FLOX_BIN" activate -- bash "$TESTS_DIR/init/node/krb5.sh"
+      # XXX "$TESTS_DIR/init/node/krb5.sh" is not always present so only run
+      #     once we have confirmed that it exists, and then expect it to fail.
+      if [ -f "$TESTS_DIR/init/node/krb5.sh" ]; then
+        run "$FLOX_BIN" activate -- bash "$TESTS_DIR/init/node/krb5.sh"
+        assert_failure
+      fi
 
       _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/resolve/krb5_after_prereqs_installed.json" \
-        "$FLOX_BIN" install krb5
+        run "$FLOX_BIN" install krb5
+      assert_success
 
-      "$FLOX_BIN" activate -- bash "$INPUT_DATA/init/node/krb5.sh"
+      run "$FLOX_BIN" activate -- bash "$INPUT_DATA/init/node/krb5.sh"
+      assert_success
       ;;
     *-darwin)
       # Ensure we're getting krb5 from the flox package by first checking
       # installation fails
-      run ! "$FLOX_BIN" activate -- bash -c 'CPATH="$FLOX_ENV/include/c++/v1:$CPATH" . "$TESTS_DIR/init/node/krb5.sh"'
+      # XXX "$TESTS_DIR/init/node/krb5.sh" is not always present so only run
+      #     once we have confirmed that it exists, and then expect it to fail.
+      if [ -f "$TESTS_DIR/init/node/krb5.sh" ]; then
+        run "$FLOX_BIN" activate -- bash -c 'CPATH="$FLOX_ENV/include/c++/v1:$CPATH" . "$TESTS_DIR/init/node/krb5.sh"'
+        assert_failure
+      fi
 
       _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/resolve/krb5_after_prereqs_installed.json" \
-          "$FLOX_BIN" install krb5
+        run "$FLOX_BIN" install krb5
+      assert_success
```

Comment:
nit: I prefer not adding the extra boilerplate of `run` and `assert_success`, although I guess we don't have a super clear convention on that. Bats exits if a command fails

Reply:
Not sure where I read it but my recollection is best practice is to use `run` and `assert_*` at every turn throughout a test rather than relying upon an implicit bash script failure. I expect it's so that bats can be more specific about exactly where a failure occurs, possibly it's better able to report line numbers that way?

Reply:
`In case you only need to check the command succeeded, it is better to not use run` https://bats-core.readthedocs.io/en/stable/writing-tests.html#when-not-to-use-run